cirq/ops/projectors.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, Dict, List, Tuple, TYPE_CHECKING
+
+import numpy as np
+
+from cirq import value
+from cirq.ops import raw_types
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@value.value_equality
+class Projector(raw_types.Gate):
+    """A non-unitary gate that projects onto a single qubit."""
+
+    def __init__(self,
+                 projection_basis: List[List[float]],
```

Comment:
Will this support numpy arrays ?

Reply:
It did not, but I modified the code so that it does now. Also added a test.